crates/oxc_linter/src/rules/eslint/prefer_numeric_literals.rs

```diff
@@ -0,0 +1,301 @@
+use oxc_ast::{
+    ast::{Argument, CallExpression, Expression},
+    AstKind,
+};
+use oxc_diagnostics::OxcDiagnostic;
+use oxc_macros::declare_oxc_lint;
+use oxc_span::Span;
+
+use crate::{
+    context::LintContext,
+    fixer::{Fix, RuleFixer},
+    rule::Rule,
+    AstNode,
+};
+
+fn prefer_numeric_literals_diagnostic(span0: Span, x0: &str) -> OxcDiagnostic {
+    OxcDiagnostic::warn(format!(
+        "eslint(prefer-numeric-literals): Use {x0} literals instead of parseInt()."
+    ))
+    .with_label(span0)
+}
+
+#[derive(Debug, Default, Clone)]
+pub struct PreferNumericLiterals;
+
+declare_oxc_lint!(
+    /// ### What it does
+    ///
+    /// Disallow parseInt() and Number.parseInt() in favor of binary, octal, and hexadecimal
+    /// literals.
+    ///
+    /// ### Why is this bad?
+    ///
+    /// The parseInt() and Number.parseInt() functions can be used to turn binary, octal, and
+    /// hexadecimal strings into integers. As binary, octal, and hexadecimal literals are supported
+    /// in ES6, this rule encourages use of those numeric literals instead of parseInt() or
+    /// Number.parseInt().
+    ///
+    /// ### Example
+    /// ```javascript
+    /// parseInt("111110111", 2) === 503;
+    /// parseInt(`111110111`, 2) === 503;
+    /// parseInt("767", 8) === 503;
+    /// parseInt("1F7", 16) === 503;
+    /// Number.parseInt("111110111", 2) === 503;
+    /// Number.parseInt("767", 8) === 503;
+    /// Number.parseInt("1F7", 16) === 503;
+    /// ```
+    PreferNumericLiterals,
+    style,
+);
+
+impl Rule for PreferNumericLiterals {
+    fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
+        if let AstKind::CallExpression(call_expr) = node.kind() {
+            match &call_expr.callee.without_parenthesized() {
+                Expression::Identifier(ident) if ident.name == "parseInt" => {
+                    if ctx.symbols().get_symbol_id_from_name("parseInt").is_none() {
```

Comment:
This code may let our rule doesn't work (Even thought I think rare people who will write code like this):

```js
// 1. Use `parseInt` here
parseInt();

{
  // 2. And then define another function called `parseInt` in this scope
  function parseInt() {
    // snip
  }
}
```

What about using `get_symbol_id_of_variable(ident, ctx).is_none()`

https://github.com/oxc-project/oxc/blob/2334515247f2db6eca56e8df4522dd11d4046057/crates/oxc_linter/src/ast_util.rs#L261-L269

Reply:
I am not sure if I understand what this does, is that an extra check to check if `parseInt` is not undefined?

Reply:
Sorry for my unclear comment, you can add this to your fail tests, the `parseInt` in first line should be detected.

```javascript
  parseInt("767", 8) === 503; // should report
  function foo() {
      function parseInt() {
          throw new Error()
      }   
  }
```